server/src/main/java/org/elasticsearch/action/search/ClearScrollController.java

```diff
@@ -166,16 +177,34 @@ public static void closeContexts(
         lookupListener.addListener(listener.delegateFailure((l, nodeLookup) -> {
             final var successes = new AtomicInteger();
             try (RefCountingRunnable refs = new RefCountingRunnable(() -> l.onResponse(successes.get()))) {
-                for (SearchContextIdForNode contextId : contextIds) {
+                for (Entry<ShardId, SearchContextIdForNode> entry : shards.entrySet()) {
+                    var contextId = entry.getValue();
                     if (contextId.getNode() == null) {
                         // the shard was missing when creating the PIT, ignore.
                         continue;
                     }
                     final DiscoveryNode node = nodeLookup.apply(contextId.getClusterAlias(), contextId.getNode());
+
+                    Set<DiscoveryNode> targetNodes;
                     if (node != null) {
+                        targetNodes = Collections.singleton(node);
+                    } else {
+                        staticLogger.info("---> missing node when closing context: " + contextId.getNode());
```

Comment:
note: We need to close the contexts after moving them when the "old" PIT is used, so if the originally encoded node it gone we try all remaining ones that currently hold that shard here (regardless of whether that node also hold a pit context.